app/javascript/dashboard/components/widgets/AutomationActionTeamMessageInput.vue

```diff
@@ -38,7 +38,15 @@ export default {
   },
   mounted() {
     const { team_ids: teamIds } = this.value;
-    this.selectedTeams = teamIds;
+    if (teamIds) {
+      if (typeof teamIds[0] !== 'object') {
+        this.selectedTeams = this.teams.filter(team =>
+          teamIds.includes(team.id)
+        );
+      } else {
+        this.selectedTeams = teamIds;
+      }
+    }
```

Comment:
As discussed during the call, the root cause of the problem will not be fixed here. Please make sure that you are passing the right values down to the component.